### ISSUE TITLE
Adding cross-ontologies relations to BP terms: GO:0042352, GO:0042351 and GO:0061619

### ISSUE THREAD

**Issue by rozaru** [geneontology/go-ontology#27681]
Hello,

Would it be possible to add the missing cross-ontologies relations to the following GO terms:

**GO:0042352**
To add:
has_primary_output CHEBI:18287 L-fucose

**GO:0042351**
To add:
has_primary_input  CHEBI:21168  GDP-mannose
has_primary_output CHEBI:57984  GDP-L-fucose(2-)

**GO:0061619**
To add:
has_primary_output CHEBI:15361  pyruvate

Thank you

Rossana

@sjm41 
@ukemi

**Comment by pgaudet** [comment]
Hi @rozaru 
Can you please tell me how you'd like to use this? We have not typically put outputs for degradation processes or inputs for synthesis processes; it's easy enough to add these three, but that seems to make the ontology inconsistent, so I'd like to understand if you specifically need these, or if we need a more global project to review what we capture WRT links to ChEBI. 

Thanks, Pascale

**Comment by rozaru** [comment]
Hi @pgaudet,

I didn't realise that it was not a standard process to add them when the boundaries of a pathway are clear. It's just that most of the specific GO terms for a pathway that I looked at they tend to have primary input/output so I thought that they were necessary for the GO term.
For example:
GDP-mannose biosynthetic process from mannose
mannose to fructose-6-phosphate catabolic process 
I don't have a usage so far for them. They are normally mention in the definition, but maybe in the future to have them in an ontology as well could help to search for them.
Mainly, I just wanted to be consistent with the other pathways. If it's causing problems, I can close the ticket.

Thanks

Rossana

**Comment by pgaudet** [comment]
The problem is that our guidelines are not exhaustive enough. 

For the logical definition of a term, we need only to include how it's different from the parent, so for example 'glucose catabolic process' is a catabolic process with primary input glucose.

With respect to start and ends, I always assumed this referred to MFs, but again the guidelines are not explicit about this. 
For example for [GO:0061621 canonical glycolysis](https://www.ebi.ac.uk/ols4/ontologies/go/classes/http%253A%252F%252Fpurl.obolibrary.org%252Fobo%252FGO_0061621), the logical definition is 

'glycolytic process through glucose-6-phosphate'
 and ('has part' some 'glyceraldehyde-3-phosphate dehydrogenase (NAD+) (phosphorylating) activity')
 and ('starts with' some 'glucokinase activity')
 and ('has primary output' some 'NADH(2-)')
 and ('has primary input' some D-glucopyranose)
 and ('has primary input' some 'NAD(1-)')

So, we have start and end reactions, but no output chemical. 
(I also understand from @cmungall that this logical definition has more in it than what is strictly necessary and sufficient, but we dont yet have clear guidelines. 


Thanks, Pascale

**Comment by sjm41** [comment]
Yes, it would definitely be helpful to have more guidelines and consistency with these logical defs.
I suppose if the main guideline is "necessary and sufficient", then there's always going to be variation in whether a LD has a 'primary input/output' or 'starts/ends with', just depending on what's needed to distinguish the term.

For GO metabolic pathway terms, it's extremely useful to have a specific (CHEBI) metabolite and/or enzyme logically defined as the start and end point in order to specify pathway boundaries and ensure consistent usage and clear meaning.

**Comment by pgaudet** [comment]
We can do this for now, but I think a less ambiguous solution would be to use 'de novo' biosynthetic process (no term yet, as far as I can tell) and GO:0043094 '(cellular) metabolic compound salvage' (Edith is currently removing 'cellular' from all these terms) as the genus, so we would have 

- GDP-fucose biosynthesis : is_a biosythetic process, has primary output some GDP-fucose 
- de novo GDP-fucose biosynthesis : is_a do novo biosythetic process, has primary output some GDP-fucose 
- GDP-fucose salvage : is_a metabolic compound salvage, has primary output some GDP-fucose 

This way is still allows is to distinguish between pathways that really do start with different compounds. I just dont think that using primary input and primary output to distinguish salvage and de novo pathways will be robust enough. 

What do you think ?

**Comment by sjm41** [comment]
That's an interesting idea. I guess we could go down that road.
Though I thought that most metabolic 'biosynthetic processes' were implicitly 'de novo biosynthetic processes' anyway - it's just that the 'de novo' prefix is explicitly added to distinguish those cases where a 'salvage' pathway is also described.
True?

I think mine and Rossana's main aim for this example (and more generally, wherever possible) is to have a clearly defined start (input) **and** end (output) as CHEBI IDs in the relationship/intersection lines.
Here, the  'de novo' GDP-L-fucose biosynthetic process term has a GO textual def, MetaCyc xref and Reactome xref that all agree on the start and end of the pathway, so it would be good if the GO term relationship/intersection clearly indicated this - then curators will be able to use such terms consistently, and computation over the annotations can happen accurately.

This is part of the wider discussion that's currently happening about how/when GO specifies the start/end of pathways - and whether that's specified as inferred/asserted classes, and whether we do this using chemicals (has_input/has_output) and/or activities (starts_with, ends_with)

**Comment by pgaudet** [comment]
> Though I thought that most metabolic 'biosynthetic processes' were implicitly 'de novo biosynthetic processes' anyway - it's just that the 'de novo' prefix is explicitly added to distinguish those cases where a 'salvage' pathway is also described.
True?

Yes, absolutely.

**Comment by pgaudet** [comment]
I guess my point is that from the LD you propose: 

intersection_of: GO:0009058 ! biosynthetic process
intersection_of: has_primary_input CHEBI:57527 ! GDP-alpha-D-mannose(2−)
intersection_of: has_primary_output CHEBI:57273 ! GDP-beta-L-fucose(2−)

I dont know how to tell that this is a de novo pathway. This is why it would be clearer to be if that were the genus.

**Comment by sjm41** [comment]
>I guess my point is that from the LD you propose
>I dont know how to tell that this is a de novo pathway. This is why it would be clearer to be if that were the genus.

Yes, good point.
But if we add a new 'de novo biosynthetic process' term for this purpose, would we then need to review all current 'biosynthetic process' terms to see which should change to using the new term as an inferred/asserted parent?
